src/api/app/models/kiwi/image.rb

```diff
@@ -143,6 +143,11 @@ def default_package_group
     package_groups.find_or_create_by(kiwi_type: :image, pattern_type: 'onlyRequired')
   end
 
+  def self.map_to_allowed_repository_types(repository)
+    repository.in?(%w(yast2 rpm-dir)) ? 'rpm-md' : repository
```

Comment:
I still don't like that there are two different kinds of not valid repository types
1) Not valid types ``yast2 rpm-dir`` which get corrected to ``rpm-md``
2) Not valid types (everything except  ``yast2 rpm-dir rpm-md apt-deb``) which will result in an parsing / build failure

So from a user perspective this feels a little bit strange. My suggestion would be to set the type to default ``rpm-md`` if the repository is not ``rpm-md apt-deb``.

@adrianschroeter what do you think?

Reply:
We could set it to rpm-md if it is not apt-deb, yes. (as long as this is only for the webui kiwi editor)